Add Get|Set-ArubaSWPort

alagoutte opened the pull request:
Add cmdlets for Get and Configure Port (name, config, status, flow control, dsnoop trust...)

alagoutte committed:
Port: Add Get-ArubaSWPort to get information about a port

Information like name/description, Configuration Mode (Speed..), status...

alagoutte committed:
Port: Add Set-ArubaSWPort to configure a port/interface

name/description, Mode Configuration, status...

alagoutte committed:
Port: Add test-integration

alagoutte committed:
Get-ArubaSWPort: Fix port_id fitler for equality search port_id

alagoutte committed:
Port(Tests): Add Tests for enable/disable a Port

alagoutte committed:
Set-ArubaSWPort: Fix example (use long parameter name)

alagoutte committed:
Port(Tests): Remove connect(ion) to -ArubaSW

alagoutte committed:
Get-ArubaSWPortStatistics for get port statistics (packets/bytes/throughtput/error TX or RX)

alagoutte committed:
Port(Tests): Add test for Get-ArubaSWPOrtStatistics cmdlet

alagoutte committed:
Port(Tests): Fix trailing whitespace

alagoutte commented:
@CedricMoreau can you look ?

CedricMoreau commented on PowerArubaSW/Public/Port.ps1:
```diff
@@ -0,0 +1,206 @@
+        Get Port of the Switch
+
+        .DESCRIPTION
+        Get Portinformation
```
Add a space ? (Between port and informtion)

CedricMoreau commented on PowerArubaSW/Public/Port.ps1:
```diff
@@ -0,0 +1,206 @@
+
+    <#
+        .SYNOPSIS
+        Get Port Sstatisticsof the Switch
```
Add a space

CedricMoreau commented on PowerArubaSW/Public/Port.ps1:
```diff
@@ -0,0 +1,206 @@
+        Get Port Sstatisticsof the Switch
+
+        .DESCRIPTION
+        Get Portinformation
```
Add a space

CedricMoreau commented on PowerArubaSW/Public/Port.ps1:
```diff
@@ -0,0 +1,206 @@
+            } else {
+                $_port | add-member -name "is_dsnoop_port_trusted" -membertype NoteProperty -Value $false
+            }
+        }
```
Add a blank line maybe ?

CedricMoreau commented on PowerArubaSW/Public/Port.ps1:
```diff
@@ -0,0 +1,206 @@
+        }
+        $response = invoke-ArubaSWWebRequest -method "PUT" -body $_port -url $url
+
+        $response | convertfrom-json
```
Add a blank line too

CedricMoreau commented on Tests/integration/Port.Tests.ps1:
```diff
@@ -0,0 +1,181 @@
+        Set-ArubaSWPort -port_id 3 -name DEFAULT_PORT
+        #Always enable the port (disabled on WorkBench...)
+        Set-ArubaSWPort -port_id 3 -is_port_enable
+    }
```
Add a blank line ?

CedricMoreau commented on Tests/integration/Port.Tests.ps1:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    It "Get ALL Port" {
+            $PORTS = Get-ArubaSWPort
```
Indent ? (not the same as below )

CedricMoreau commented on Tests/integration/Port.Tests.ps1:
```diff
@@ -0,0 +1,181 @@
+#TODO: Add check if no ipaddress/login/password info...
+
+Describe  "Get Port" {
+    BeforeAll {
```
Add a blank line (or remove it at line 43)

CedricMoreau commented on Tests/integration/Port.Tests.ps1:
```diff
@@ -0,0 +1,181 @@
+        $PORT.drop_tx | Should -Not -BeNullOrEmpty
+        $PORT.port_speed_mbps | Should -Not -BeNullOrEmpty
+    }
+}
```
Add a blank line ?

CedricMoreau commented on Tests/integration/Port.Tests.ps1:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    It "Get ALL Port Stastistics" {
+            $PORTS = Get-ArubaSWPortStatistics
```
Indent ?

alagoutte committed:
Port: Fix typo/indent after review

alagoutte committed:
README(.md): Add new Ports cmdlets and on feature list

2 new Get (Get-ArubaSWPort/Get-ArubaSWPortStatistics) and a Set (Set-ArubaSWPort) cmdlets
